iohk/NixOps.hs

```diff
@@ -423,13 +454,19 @@ deploy o c@NixopsConfig{..} evonly buonly = do
      unless keyExists $
        die "Deploying nodes, but 'keys/key1.sk' is absent."
 
+  printf ("Generating 'cluster.nix' from '"%fp%"'..\n") cCluster
+  exists <- testpath cCluster
+  unless exists $
+    die $ format ("Cluster config '"%fp%"' doesn't exist.") cCluster
+  inproc "yaml2json" [format fp cCluster] empty & output "cluster.nix"
+
   printf ("Deploying cluster "%s%"\n") cName
   export "NIX_PATH_LOCKED" "1"
   export "NIX_PATH" (nixpkgsCommitPath cNixpkgsCommit)
   when (not evonly) $ do
     when (elem Nodes cElements) $ do
       export "GC_INITIAL_HEAP_SIZE" (showT $ 8 * 1024*1024*1024) -- for 100 nodes it eats 12GB of ram *and* needs a bigger heap
-    export "SMART_GEN_IP"     =<< incmd o "curl" ["--silent", fromURL awsPublicIPURL]
+    export "SMART_GEN_IP"     =<< getIP <$> deployerIP o
```

Comment:
Do we really still need to export this? Reference in `modules/cardano-node.nix:` could just use a setting now.